draft-ietf-ppm-dap.md

```diff
@@ -1356,15 +1353,16 @@ This message consists of:
 * `part_batch_selector`: The "partial batch selector" used by the Aggregators to
   determine how to aggregate each report:
 
-    * For `fixed_size` tasks, the Leader specifies a "batch ID" that determines
-      the batch to which each report for this aggregation job belongs.
+    * For `leader_selected` tasks, the Leader specifies a "batch ID" that
+      determines the batch to which each report for this aggregation job
+      belongs.
 
-      [OPEN ISSUE: For fixed_size tasks, the Leader is in complete control over
-      which batch a report is included in. For time_interval tasks, the Client
-      has some control, since the timestamp determines which batch window it
-      falls in. Is this desirable from a privacy perspective? If not, it might
-      be simpler to drop the timestamp altogether and have the agg init request
-      specify the batch window instead.]
+      [OPEN ISSUE: For leader_selected tasks, the Leader is in complete control
+      over which batch a report is included in. For time_interval tasks, the
+      Client has some control, since the timestamp determines which batch window
+      it falls in. Is this desirable from a privacy perspective? If not, it
+      might be simpler to drop the timestamp altogether and have the agg init
+      request specify the batch window instead.]
```

Comment:
This doesn't appear to be desirable from a privacy perspective. I'll note here that having the timestamp in the metadata seems to be helpful for replay protection, since it allows you to shard by batch interval.

I suggest we just delete this text.

Reply:
I'm sure we've discussed this before so maybe I am missing something but there is an attack where a Leader separately collects reports from a single client and then only aggregates those.

For time intervals the leader is strongly constrained here and that's exactly why we prohibit overlapping time-intervals. We might however have lost this meaning in the wording over time...

> If the report pertains to a batch that was previously collected, then the input share MUST be marked as invalid with error batch_collected.

> Finally, the Aggregator checks that the batch does not contain a report that was included in any previous batch. If this batch overlap check fails, then the Aggregator MUST abort with error of type "batchOverlap". For time_interval tasks, it is sufficient (but not necessary) to check that the batch interval does not overlap with the batch interval of any previous query. If this batch interval check fails, then the Aggregator MAY abort with error of type "batchOverlap".

> For each report in the batch, the time at which that report was generated (see [Section 4.4](https://ietf-wg-ppm.github.io/draft-ietf-ppm-dap/draft-ietf-ppm-dap.html#upload-flow)) MUST fall within the batch interval specified by the Collector.

According to my probably incorrect memory this is supposed to be prevented by the collector deciding which reports to aggregate and the other aggregator checking as well.
We should be explicit about this attack in https://ietf-wg-ppm.github.io/draft-ietf-ppm-dap/draft-ietf-ppm-dap.html#name-batch-parameters

Reply:
Deleting the text SGTM because I think `time_interval`'s association of report to batch via timestamp is (plausibly) desirable from a privacy perspective.

Reply:
@simon-friedberger I agree that time interval queries are much more constrained, and this definitely has benefits. But I don't think fixed size queries are unconstrained in a way that leads to overlapping batches. The aggregators are supposed to assign reports to batches and make sure each report is mapped to a single batch. At a high level, any query type needs to ensure that queries _partition_ the set of reports for a task.

Reply:
I added some text to the security considerations section capturing this concern.